grammar.js

```diff
@@ -297,7 +298,23 @@ module.exports = grammar(HTML, {
       '}',
     ),
 
-    attribute_name: _ => /[^<>{}"'/=\s]+/,
+    attribute_directive: _ => /[^<>{}:|"'/=\s]+/,
+
+    attribute_identifier: _ => /[^<>{}:|"'/=\s]+/,
+
+    modifier: _ => /[^<>{}:|"'/=\s|]+/,
+
+    attribute_name: $ => seq(
+      choice(
+        prec(1, $.attribute_identifier),
+        seq(
+          $.attribute_directive,
+          ':',
+          $.attribute_identifier,
+        ),
+      ),
+      repeat(seq('|', $.modifier)),
```

Comment:
I'm sure we could bikeshed the actual node names for days, but maybe `modifier` could be `attribute_modifier` to match the others?

Reply:
Done